### PR TITLE
Refactor song menu

### PR DESCRIPTION
**Pull request by el-dasher**
Basically i followed my IDE to try to mitigate as many warns as possible in the menu classes, also simplified a piece of code in ModMenu.java, till now i didn't found any weird behaviours.

### CODE REVIEW

**Comment by Rian8337** [comment]
Have you tested this? Sometimes refactor give undesired changes to functionalities.

**Review by Rian8337 — COMMENTED** [review]
I think this needs further testing for online functionalities. I will do it later.

On the other hand, I have a few questions be answered, but keep in mind that this is a huge PR, so I may have missed a few details.

File: src/ru/nsu/ccfit/zuev/osu/menu/ScrollBar.java [hunk 1]
```diff
@@ -9,10 +9,9 @@
 
 public class ScrollBar {
     private final Rectangle barRectangle;
-    private float maxY = 0;
     private boolean visible;
 
-    public ScrollBar(final Scene scene, final IScrollBarListener listener) {
+    public ScrollBar(final Scene scene, SongMenu songMenu) {
```
[the remark this answers is not in the document]
Comment (el-dasher) [reply src/ru/nsu/ccfit/zuev/osu/menu/ScrollBar.java:14]: yes, some stuff i didn't dele5ed cause i couldn't figured out how it was used in the firat place.

**Comment by el-dasher** [comment]
ok

File: src/ru/nsu/ccfit/zuev/osu/menu/ScrollBar.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (el-dasher) [reply src/ru/nsu/ccfit/zuev/osu/menu/ScrollBar.java]: I tested querying beatmaps, sorting beatmaps apply mods, change settings and tested pausing and others and couldn't find any weird behaviour.

**Comment by el-dasher** [comment]
I tested the menus and applying settings etc and as it seens it's working properly, including querying and sorting beatmaps.

**Review by Rian8337 — APPROVED** [review]

**Pull request merged**